lib/chef/knife/vsphere_vm_network_add.rb

```diff
@@ -50,19 +60,65 @@ def run
     else fail
     end
 
-    vm.ReconfigVM_Task(
-      spec: {
-        deviceChange: [{
-          operation: :add,
-          fileOperation: nil,
-          device: RbVmomi::VIM::VirtualVmxnet3(
-            key: -1,
-            deviceInfo: { summary: summary, label: '' },
-            backing: backing,
-            addressType: 'generated'
-          )
-        }]
-      }
-    ).wait_for_completion
+    adapter_type = ''
+    if config[:adapter_type] == 'vmxnet3'
+      puts 'Adapter type: vmxnet3'
```

Comment:
I think I'd prefer dropping all the `puts`, or at the very least, adding `if log_verbose?`. If you use the `get_config` method above, if the user runs this with the `-V` flag they'll see all the options they entered.